commit e79d6bb1d272100fb78d54ce60e9c5f5e24ea8cd (refs/changes/24/1624/3)
Author: Marsell Kukuljevic <marsell@joyent.com>
Date:   2017-03-15T12:16:31+13:00 (2 years, 7 months ago)
    
    PUBAPI-1368 - CloudAPI can die on MethodNotAllowed during start up
    Reviewed by: Julien Gilli <julien.gilli@joyent.com>
    Reviewed by: Pedro P. Candel <pedro@joyent.com>

diff --git a/main.js b/main.js
index 60812b9..9c2f3f4 100644
--- a/main.js
+++ b/main.js
@@ -168,17 +168,18 @@ function createBootstrapServer(port, cb) {
 
     function bootstrapHandler(req, res, next) {
         res.send(new restify.InternalError('Failure to connect to Moray'));
+        next();
+    }
 
-        if (next) {
-            next();
-        }
+    function bootstrapMethodNotAllowed(req, res, err, next) {
+        bootstrapHandler(req, res, next);
     }
 
     ['get', 'post', 'put', 'del', 'head'].forEach(function (method) {
         bootstrapServer[method]('/.*/', bootstrapHandler);
     });
 
-    bootstrapServer.on('MethodNotAllowed', bootstrapHandler);
+    bootstrapServer.on('MethodNotAllowed', bootstrapMethodNotAllowed);
 
     bootstrapServer.listen(port, function () {
         cb(null, bootstrapServer);
